e2e/include/idata.h

```diff
@@ -23,84 +25,100 @@
 // Individual data used to store expected values from astarte
 typedef struct
 {
-    // the buffer will be freed by the idata free function
     const char *path;
-    // the buffer will be freed by the idata free function
     astarte_data_t data;
     e2e_timestamp_option_t timestamp;
-} e2e_individual_data_t;
+} idata_individual_t;
 
 // Property data used to store expected values from astarte
 // if unset is true the data element will contain an invalid value
 typedef struct
 {
-    // the buffer will be freed by the idata free function
     const char *path;
-    // the buffer will be freed by the idata free function
     astarte_data_t data;
     bool unset;
-} e2e_property_data_t;
+} idata_property_t;
 
 // Object data used to store expected values from astarte
 typedef struct
 {
-    // the buffer will be freed by the idata free function
     const char *path;
-    // the buffer will be freed by the idata free function
     e2e_object_entry_array_t entries;
-    // the buffer will be freed by the idata free function
     e2e_byte_array object_bytes;
     e2e_timestamp_option_t timestamp;
-} e2e_object_data_t;
+} idata_object_t;
 
-// Data expected in the e2e tests, this stuct identifies an atomic received piece of data
-// In case of an individual or property interface this refers to a single individual value
-// In case of an object interface to the whole aggregate
-// The union follows the type and aggregation of the astarte interface struct
-typedef struct
+typedef union
 {
-    const astarte_interface_t *interface;
-    sys_dnode_t node;
-    union
-    {
-        e2e_individual_data_t individual;
-        e2e_property_data_t property;
-        e2e_object_data_t object;
-    } values;
-} e2e_idata_unit_t;
-
-void idata_unit_log(e2e_idata_unit_t *idata_unit);
-
-// FIXME for now the order of insertion matters, it has to be the same as the order in which
-// messages are received for now it is fixed by giving the user the ability to iterate through the
-// matches still i could choose a better alternative
+    idata_property_t property;
+    idata_individual_t individual;
+    idata_object_t object;
+} astarte_message_t;
+
+// FIXME replace struct definition with with SPSC_DECLARE once they remove the static
```

Comment:
```suggestion
// TODO replace struct definition with with SPSC_DECLARE once they remove the static
```
I would make a to-do as it's not really a fix.